org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,367 @@
+	private Object oldTableID;
+	
+	private GridField tableIDGridField;
+	private WEditor tableIDEditor;
+	private GridTabDataBinder dataBinder;
+	
+	private Div wrapperDiv;
+	private Textbox recordTextBox;
+	private ToolBarButton editButton;
+	private ToolBarButton zoomButton;
+	
+	protected String selectedTableName = "";
+	protected String selectedRecordIdentifyer = "";
+	
+	private static final String IMAGES_CONTEXT_ZOOM_PNG = "images/Zoom16.png";
+	private static final String IMAGES_CONTEXT_EDIT_RECORD_PNG = "images/EditRecord16.png";
+
+	public WRecordIDEditor(GridField gridField, boolean tableEditor, IEditorConfiguration editorConfiguration) {
+		super(new Vlayout(), gridField, tableEditor, editorConfiguration);
+		init();
+	}
+
+	private void init() {
+		wrapperDiv = new Div();
+		wrapperDiv.setSclass("recordid-editor");
+		wrapperDiv.setParent(this.getComponent());
+		wrapperDiv.addEventListener(Events.ON_RIGHT_CLICK, this);
+
+		if(gridTab != null) {
+			tableIDGridField = gridTab.getField("AD_Table_ID");
+			tableIDEditor = WebEditorFactory.getEditor(tableIDGridField, true);
+			
+			if(tableIDGridField.getValue() != null)
+				tableIDEditor.setValue(tableIDGridField.getValue());
+						
+			dataBinder = new GridTabDataBinder(gridTab);
+			tableIDEditor.addValueChangeListener(dataBinder);
+			
+			tableIDGridField.addPropertyChangeListener(tableIDEditor);
+			tableIDGridField.addPropertyChangeListener(new PropertyChangeListener() {
+				@Override
+				public void propertyChange(PropertyChangeEvent evt) {
+					if(evt.getNewValue() != oldTableID) {
+						oldTableID = evt.getNewValue();
+						setValue(null);
+					}
+				}
+			});
+			oldTableID = tableIDGridField.getValue();
+		}
+		
+		recordTextBox = new Textbox();
+		recordTextBox.setParent(wrapperDiv);
+		recordTextBox.setWidth("100%");
+		recordTextBox.setReadonly(true);
+		
+		editButton = new ToolBarButton();
+		editButton.setStyle("right: 21px;");
+		if (ThemeManager.isUseFontIconForImage())
+			editButton.setIconSclass("z-icon-Edit");
+		else
+			editButton.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_EDIT_RECORD_PNG));
+		editButton.setParent(wrapperDiv);
+		editButton.addEventListener(Events.ON_CLICK, this);
+
+		zoomButton = new ToolBarButton();
+		if (ThemeManager.isUseFontIconForImage())
+			zoomButton.setIconSclass("z-icon-Zoom");
+		else
+			zoomButton.setImage(ThemeManager.getThemeResource(IMAGES_CONTEXT_ZOOM_PNG));
+		zoomButton.setParent(wrapperDiv);
+		zoomButton.addEventListener(Events.ON_CLICK, this);
+		
+		if (gridField != null)
+        {
+			popupMenu = new WEditorPopupMenu(true, true, isShowPreference());
+			popupMenu.addMenuListener(this);
+			addChangeLogMenu(popupMenu);
+        }
+	}
+	
+	public void actionRefresh(Object value)
+    {
+		setValue(value);
+    }
+	
+	@Override
+	public void actionZoom()
+	{
+		Integer tableID = (Integer) tableIDGridField.getValue();
+		Integer recordID = (Integer) gridTab.getValue("Record_ID");
```

Comment:
becarefull with such casting, you will get NPE if either AD_Table_ID or Record_ID value is null.